MASFoundation/Classes/_private_/models/MASWebSessionBrowserBasedAuthentication.h

```diff
@@ -18,4 +24,23 @@
 API_AVAILABLE(ios(12.0), macCatalyst(13.0), macos(10.15), watchos(6.2))
 @interface MASWebSessionBrowserBasedAuthentication : NSObject <MASTypedBrowserBasedAuthenticationInterface>
 
+
+///--------------------------------------
+/// @name Lifecycle
+///--------------------------------------
+
+# pragma mark - Lifecycle
+
+/**
+ * Initializer to perform default initialization.
+ *
+ * @param callbackURLScheme NSString containing the callback url scheme to use.
+ * @return Returns the newly initialized MASWebSessionBrowserBasedAuthentication.
+ */
+- (instancetype)initWithCallbackURLScheme:(NSString *)callbackURLScheme;
```

Comment:
** Looks all good to me.
But had a bit of a hitch with taking an input for a custom url scheme from the customer.
I am coming from the perspective of 'its a SDK/Library', we might don't want to ask for the custom url scheme of the app.
What i am trying to see here is, we open the web login template on the ASWebAuthSession and upon the completion it redirects to the custom url scheme, this kicks the app re-opened again and the app delegate, delegates the callback url to the AuthResponse class. Is it possible through ASWebSession ??

Reply:
I removed the need to provide the url scheme altogether for the WebSession browser authentication type. This will be extracted from the MSSO config.